src/app/info/ClientSideHelpers.tsx

```diff
@@ -1,18 +1,34 @@
 "use client";
 
-import { RoleUpdateContext } from "@/app/roles";
-import { useContext } from "react";
+import { DbWikiRow } from "@/databaseUtils";
+import { useState } from "react";
+import DefaultWikiItemView from "@/app/info/DefaultWikiItemView";
+import EditWikiItemView from "@/app/info/EditWikiItemView";
 
-export const buttonAlert = (): void => {
-    alert("Wiki item has been updated!");
+export const enterEditMode = (setIsInEditMode: (isInEditMode: boolean) => void): void => {
+    setIsInEditMode(true);
 };
 
-interface RoleProps {
-    children?: React.ReactNode;
+interface EditProps {
+    isInEditMode: boolean;
+    row?: DbWikiRow;
 }
 
-export const AdminManagerDependent: React.FC<RoleProps> = (props) => {
-    const { role } = useContext(RoleUpdateContext);
-
-    return <>{(role === "admin" || role === "manager") && props.children}</>;
+export const EditModeDependentItem: React.FC<EditProps> = ({ isInEditMode: isInEditMode, row }) => {
+    const [rowRef, setRowRef] = useState<DbWikiRow | undefined>(row);
+    const [isInEditModeRef, setIsInEditModeRef] = useState(isInEditMode);
```

Comment:
Refs are something very specific in react, can we avoid using it in names of things that aren't refs? I think
```
    const [rowData, setRowData] = useState<DbWikiRow | undefined>(row);
    const [isInEditMode, setIsInEditMode] = useState(isInEditMode);
```
Should be fine